AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2236,6 +2238,31 @@ - (void)setDisplaySuspended:(BOOL)flag
   }
 }
 
+static const char *ASDisplayNodeDrawingPriorityKey = "ASDrawingPriority";
+
+- (void)setDrawingPriority:(NSUInteger)drawingPriority
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
+  if (drawingPriority == ASDefaultDrawingPriority) {
+    _flags.hasCustomDrawingPriority = NO;
+    objc_setAssociatedObject(self, ASDisplayNodeDrawingPriorityKey, nil, OBJC_ASSOCIATION_ASSIGN);
+  } else {
+    _flags.hasCustomDrawingPriority = YES;
```

Comment:
I think you did a great job of implementing this efficiently.  However, could you explain why you chose to prefer associated objects?  I'm curious if it is possible to just have a property for the number, which would help when we start to use this more heavily, e.g. to always prioritize text rendering.